src/arrayMethodJoin.js

```diff
@@ -5,7 +5,34 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let separatorValue = separator;
+
+    if (separatorValue === undefined) {
+      separatorValue = ',';
+    }
+
+    const arr = this;
```

Comment:
I'd say `arr` variable is unnecessary, from the context of this implementation it's implied this refers to an array